src/rules/flake8_pie/rules.rs

```diff
@@ -106,6 +106,53 @@ pub fn dupe_class_field_definitions<'a, 'b>(
     }
 }
 
+// PIE796
+pub fn prefer_unique_enums<'a, 'b>(checker: &mut Checker<'a>, parent: &'b Stmt, body: &'b [Stmt])
+where
+    'b: 'a,
+{
+    let StmtKind::ClassDef { bases, .. } = &parent.node else {
+        return;
+    };
+
+    let is_enum = bases.iter().any(|stmt| match &stmt.node {
+        ExprKind::Name { id, .. } => id == "Enum",
+        ExprKind::Attribute { value, attr, .. } => {
+            attr.ends_with("Enum")
+                && match &value.node {
+                    ExprKind::Name { id, .. } => id == "enum",
+                    _ => false,
+                }
+        }
+        _ => false,
+    });
+
+    if !is_enum {
+        return;
+    }
+
+    let mut seen_targets: FxHashSet<String> = FxHashSet::default();
+    for stmt in body {
+        let target = match &stmt.node {
```

Comment:
Could we instead use...

```rs
let comparable_value: ComparableExpr = (&value).into();
if !seen_targets.insert(comparable_value) {
  ...
}
```

Not exact code, just pseudo code. But this would detect any duplicate values, not just constants.

If we _do_ want to leave it as constants, can we change the `ExprKind::Constant` match to use `ExprKind::Constant { value: Constant::Str(value), .. }`? And remove the `value.to_string()`? That way, we'll _only_ match on strings, which I think better matches the desired semantics.

Reply:
good one, i didn't know about **ComparableExpr**  😅. It makes everything a bit more simple.